app/services/state_file/send_post_deadline_reminder_service.rb

```diff
@@ -0,0 +1,39 @@
+module StateFile
+  class SendPostDeadlineReminderService
+    BATCH_SIZE = 10
+    HOURS_AGO = 24
+
+    def self.run
+      cutoff_time_ago = HOURS_AGO.hours.ago
+      intakes_to_notify = []
+
+      ApplicationRecord::STATE_INTAKE_CLASS_NAMES.each do |base_class|
+        class_object = base_class.constantize
+        intakes_to_notify += class_object.left_joins(:efile_submissions)
+                                         .where(efile_submissions: { id: nil })
+                                         .where("#{base_class.underscore.pluralize}.created_at < ?", cutoff_time_ago)
+                                         .where.not(email_address: nil)
+                                         .where.not(email_address_verified_at: nil)
+                                         .where(unsubscribed_from_email: false)
+                                         .where("#{base_class.underscore.pluralize}.message_tracker #> '{messages.state_file.post_deadline_reminder}' IS NULL")
```

Comment:
Is this needed? I thought the `send_only_once?` config would take care of this.

Reply:
100% not needed; added for performance.  Since intakes that have this message sent already anyway, I don't see why we should clog up system resources in DelayedJobs for emails that won't get sent anyway.  Thinking forward for next years, this could be a TON of intakes that get loaded into memory for no reason.  Thanks for bringing it up though.